zookeeper/core/component_test.py

```diff
@@ -132,23 +132,16 @@ class GrandParent:
         grand_parent,
         {
             "a": 10,
-            "parent.a": 15,
```

Comment:
The tests in this file had to be modified because they were previously relying on the behaviour that this PR is explicitly disallowing.